semver.go

```diff
@@ -27,19 +27,22 @@ var cleanup = strings.NewReplacer("  ", " ", "> ", ">", "= ", "=", "< ", "<")
 
 //strips any prefix for the version for parsing
 func stripPrefix(version string) string {
-	switch version[0] {
-	case '>', '<':
-		if version[1] == '=' {
-			version = version[2:]
-		} else {
+	// Avoiding index out of range errors.
+	if len(version) > 0 {
```

Comment:
Instead of guarding each if block, let's just add a check and short-circuit at the top of the function.

```go
if len(version) == 0 {
  return ""
}
```

Reply:
Thanks for the response. Hmm yea I was being a bit lazy because a single length check wouldn't suffice here but I see what you are getting at. I think a smarter check that short circuits would be much more elegant.

Reply:
Let me know what you think of this. Making the short circuit simple was difficult because of all the scenarios where it would panic so I just simplified it in general.

aae04f4922324496e5bfe181e5a4ae9e73e8cc3e

Reply:
Whoops didn’t see the above reply. Let me know if you want me to revert. Great idea

Reply:
I guess it depends which one is clearer.

With the regex route we don't even need the function, we should:
- change the name of the regex to `stripPrefixRx` and out of the function to package level. That way we can remove the function, and doing the `MustCompile` outside will cause a panic at startup (if the regex is invalid) rather than when/if that function is used making it far more likely to detect a problem in the future.
- then call `stripPrefixRx.ReplaceAllString` instead of `stripPrefix`.
- The regex would remove the second equal in `==`, I think this would work though: `^([=^~]|[><]=?)?v?`

Reply:
Good call. Thanks man e478bb6a27c110418f240525d84c0c013e181cee